src/silverlabnwb/generate_extended_schema.py

```diff
@@ -0,0 +1,48 @@
+from pynwb.spec import NWBAttributeSpec, NWBDatasetSpec, NWBGroupSpec, NWBNamespaceBuilder
+
+
+def generate_extended_schema():
+    ns_builder = NWBNamespaceBuilder('Extensions for acousto-optic lens data', 'silverlab_extended_schema',
+                                     'Silverlab data extension to NWB format for acousto-optic lens experiments',
+                                     version='0.1')
+    ns_builder.include_type('NWBDataInterface', namespace='core')
+
+    # Silver lab optophysiology extension
+    # attributes
+    cycle_time_attr = NWBAttributeSpec('cycle_time', 'value for cycle time', 'float')
```

Comment:
Can we make our documentation of these properties a bit more descriptive? For instance:
* cycle_time: the time in seconds for the microscope to acquire all ROIs once and return to its starting position
* cycles_per_trial: how many microscope cycles occur in each experimental trial
* imaging_mode: the acquisition mode for the experiment; pointing = single-voxel ROIs, miniscan = 2d rectangular ROIs, volume = 3d cuboid ROIs
* frame_size: the 2d imaging frame size in voxels

Of these, imaging_mode and frame_size could be considered part of the experimental setup so conceptually should probably be in /general/optophysiology if I'm reading the spec right. Whereas cycle_time and cycles_per_trial seem more suited to /processing - they could be calculated from the raw data.

Despite the fact that the Labview export itself calls the dataset zplane_pockels, I'm not sure it's the best name for it! But probably we shouldn't change that without asking the researchers... We could document it in our extension as recording microscope calibration data for focusing at different planes along the z axis though?

        The .dat file has 4 columns: Z offset from focal plane (micrometres), normalised Z,
        'Pockels' i.e. laser power in %, and z offset for drive motors.

This means it is conceptually really part of the experimental setup / imaging plane definition.

How hard is it now to define an extension that sits somewhere in /general, maybe extending ImagingPlane? (Though you'd probably want to stick one row of the pockels table in each imaging plane, so gets a bit tortuous.) As @ageorgou says, while the silverlab_api_version would definitely be LabMetaData, I don't think the rest is (maybe imaging_mode? not sure!). Being pragmatic, if we have it working to store in /processing, that'll do for this PR, as it's fairly easy to move later (providing we do so before the first release that the group use in anger).